test/activations.jl

```diff
@@ -7,9 +7,27 @@
 end
 
 @testset "forward diff" begin
-    f(x) = logσ.(x)
+    f = (x) -> logσ.(x)
     ds = Dual.(rand(5),1)
     @test f(ds) ≈ collect(f(CuArray(ds)))
+    f = (x) -> tanh.(x)
+    ds = Dual.(rand(5),1)
+    @test f(ds) ≈ collect(f(CuArray(ds)))
+    f = (x) -> σ.(x)
+    ds = Dual.(rand(5),1)
+    @test f(ds) ≈ collect(f(CuArray(ds)))
+    f = (x) -> elu.(x)
+    ds = Dual.(rand(5),1)
+    @test f(ds) ≈ collect(f(CuArray(ds)))
+    f = (x) -> relu.(x)
+    ds = Dual.(rand(5),1)
+    @test f(ds) ≈ collect(f(CuArray(ds)))
```

Comment:
I would drop the anonymous function entirely and turn this into a loop over activation functions.

Reply:
I've copied the activation broadcast test syntax (although I don't entirely understand the need for the @eval and interpolation), so the Dual tests now cover all activations accept rrelu (which I'm guessing doesn't work on gpu?).

Could it be worth adding similar tests to NNlib itself? I noticed it lacked any Dual number tests.

Reply:
`@eval` is needed because `NNlib.ACTIVATIONS` is an [array of symbols](https://github.com/FluxML/NNlib.jl/blob/ff3ac6eb807e9b41f46f28f8b3287d19f4b722c7/src/activations.jl#L6-L12). As for `rrelu`, it appears to be excluded because it is [non-deterministic](https://github.com/FluxML/NNlib.jl/blob/ff3ac6eb807e9b41f46f28f8b3287d19f4b722c7/src/activations.jl#L260).